common/src/main/java/org/conscrypt/NativeCrypto.java

```diff
@@ -52,13 +68,39 @@
 @Internal
 public final class NativeCrypto {
     // --- OpenSSL library initialization --------------------------------------
+    private static final Throwable UNAVAILABILITY_CAUSE;
     static {
-        NativeCryptoJni.init();
-        clinit();
+        Throwable cause = null;
```

Comment:
Another PR seems reasonable. I guess any of the core classes that can be publicly instantiated would need the check ... would there be anything else?